core/src/main/java/com/linecorp/armeria/server/auth/AuthorizationStatus.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.auth;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * Provides a status of the request authorization operation, optionally combined with {@link AuthSuccessHandler}
+ * and {@link AuthFailureHandler} to facilitate custom status handling.
+ */
+@UnstableApi
+public interface AuthorizationStatus {
+
+    /**
+     * Creates {@link AuthorizationStatus} based on boolean status.
+     */
+    static AuthorizationStatus of(boolean status) {
+        return new AuthorizationStatusImpl(status);
+    }
+
+    /**
+     * Creates {@link AuthorizationStatus} based on boolean status.
```

Comment:
Could you explain what `null` means and when `null` is returned in Javadoc?

Reply:
I've used the Nullable signature
`@Nullable static AuthorizationStatus of(@Nullable Boolean status)`
to deal with conversion of binary status and `AuthorizationStatus`. Since the old-style authorization returned  `CompletableFuture<Boolean>` which could could be effectively NULL, this conversion allows to keep the processing of the result inside the `AuthService` as below:
```
                if (cause == null) {
                    if (result != null) {
                        return result.status() ? ...
```

Reply:
I see. If this is not meant to be used by a user, how about hiding it somewhere, such as `AuthService`?

Reply:
Removed this method from `AuthorizationStatus` and moved the `NULL` check to the `Authorizer`